ddoc-extraction.js

```diff
@@ -0,0 +1,91 @@
+var db = require('./db'),
+    async = require('async'),
+    _ = require('underscore');
+
+var getMasterDdoc = function(callback) {
+  db.medic.get('_design/medic', function(err, ddoc) {
+    // explitly call the callback with just these two params. Couch passes more
+    // params to the callback, and we don't want those feeding into async.waterfall
+    callback(err, ddoc);
+  });
+};
+
+var extractDdocs = function(ddoc, callback) {
+  var attachmentNames = Object.keys(ddoc._attachments).filter(function(name) {
+    // return name.startsWith('ddocs/compiled/'); // for nodejs > 0.12
+    return name.indexOf('ddocs/compiled/') === 0;
+  });
+
+  console.log('Found ' + attachmentNames.length + ' attached ddocs');
+
+  callback(null, ddoc._rev, attachmentNames);
+};
+
+var ddocNameFromAttachmentName = function(attachmentName) {
+  return '_design/' + /ddocs\/compiled\/(.+)\.json/.exec(attachmentName)[1];
+};
+
+var updateIfRequired = function(masterRevision, attachmentName, callback) {
+  var ddocName = ddocNameFromAttachmentName(attachmentName);
+
+  console.log('Looking at', attachmentName);
+
+  db.medic.get(ddocName, function(ddocErr, ddoc) {
+    if (ddocErr && ddocErr.error !== 'not_found') {
+      callback(ddocErr);
+    } else {
+      db.medic.get('_design/medic/'+attachmentName, function(attachErr, attachedDdoc) {
+        if (attachErr) {
+          callback(attachErr);
+        } {
```

Comment:
Missing `else`?

I like the style...

```
if (err) {
   return callback(err);
}
// non-error code
```
This keeps the code less indented, and short circuits on err immediately.